docs/_site/docs/templates/design.html

```diff
@@ -0,0 +1,364 @@
+    `</span><span class="p">;</span>
+  <span class="p">}</span>
+  <span class="nx">updated</span><span class="p">(){</span>
+    <span class="nx">console</span><span class="p">.</span><span class="nx">log</span><span class="p">(</span><span class="s1">'Updated'</span><span class="p">);</span>
+  <span class="p">}</span>
+  <span class="nx">changeMe</span><span class="p">(</span><span class="nx">prop</span><span class="p">){</span>
+    <span class="kd">var</span> <span class="nx">rand</span> <span class="o">=</span> <span class="nb">Math</span><span class="p">.</span><span class="nx">floor</span><span class="p">(</span><span class="nb">Math</span><span class="p">.</span><span class="nx">random</span><span class="p">()</span><span class="o">*</span><span class="mi">100</span><span class="p">);</span>
+    <span class="nx">prop</span><span class="o">==</span><span class="s1">'prop1'</span><span class="p">?</span><span class="k">this</span><span class="p">.</span><span class="nx">prop1</span><span class="o">=</span><span class="nx">rand</span><span class="p">:</span><span class="k">this</span><span class="p">.</span><span class="nx">prop2</span><span class="o">=</span><span class="nx">rand</span><span class="p">;</span>
+  <span class="p">}</span>
+<span class="p">}</span>
+<span class="nx">customElements</span><span class="p">.</span><span class="nx">define</span><span class="p">(</span><span class="s1">'my-element'</span><span class="p">,</span> <span class="nx">MyElement</span><span class="p">);</span>
+
+</code></pre></div></div>
+
+<project-displayer folder="/projects/docs/templates1" openFile="my-element.js" forceEmbedLayout="" view="editor">
+<stack-blitz slot="embed"></stack-blitz>
+</project-displayer>
+
+<h3 id="design-templates-as-a-function-of-properties">Design templates as a function of properties</h3>
+
+<p>To maximize the benefits of using the LitElement library, we recommend that you design your element’s template as a pure function of its properties.</p>
+
+<p>This means that an element’s <code class="highlighter-rouge">render</code> function</p>
+
+<ul>
+  <li>should not change the element’s state.</li>
+  <li>should not have side effects.</li>
+  <li>should not depend on anything except the element’s properties.</li>
+  <li>should return the same result when given the same inputs; that is, with the same set of property values, you should get the same template.</li>
+</ul>
+
+<p>We also recommend that you avoid manipulating DOM. Instead, express the element’s template as a function of its state, and capture element state in properties.</p>
+
+<h3 id="example">Example</h3>
+
+<p>The following element uses inefficient DOM manipulation:</p>
+
+<p><em>dom-manip.js</em></p>
+
+<div class="language-text highlighter-rouge"><div class="highlight"><pre class="highlight"><code>import { LitElement, html } from '@polymer/lit-element';
+
+/**
+ * Anti-pattern. Avoid manipulating DOM.
```

Comment:
*manually*

Reply:
Lit-html manipulates the DOM, so the anti-pattern is that people manually manipulate it